src/TestProblem/Hydro/AdvectTracers/Par_Init_ByFunction_AdvectTracers.cpp

```diff
@@ -0,0 +1,106 @@
+//                   --> Currently it only loads particle mass, position x/y/z, and velocity x/y/z
+//                       (and exactly in this order)
+//
+// Parameter   :  NPar_ThisRank : Number of particles to be set by this MPI rank
+//                NPar_AllRank  : Total Number of particles in all MPI ranks
+//                ParMass       : Particle mass     array with the size of NPar_ThisRank
+//                ParPosX/Y/Z   : Particle position array with the size of NPar_ThisRank
+//                ParVelX/Y/Z   : Particle velocity array with the size of NPar_ThisRank
+//                ParTime       : Particle time     array with the size of NPar_ThisRank
+//                ParType       : Particle type     array with the size of NPar_ThisRank
+//                AllAttribute  : Pointer array for all particle attributes
+//                                --> Dimension = [PAR_NATT_TOTAL][NPar_ThisRank]
+//                                --> Use the attribute indices defined in Field.h (e.g., Idx_ParCreTime)
+//                                    to access the data
+//
+// Return      :  ParMass, ParPosX/Y/Z, ParVelX/Y/Z, ParTime, AllAttribute
+//-------------------------------------------------------------------------------------------------------
+
+void Par_Init_ByFunction_AdvectTracers( const long NPar_ThisRank, const long NPar_AllRank,
+                                        real *ParMass, real *ParPosX, real *ParPosY, real *ParPosZ,
+                                        real *ParVelX, real *ParVelY, real *ParVelZ, real *ParTime,
+                                        real *ParType, real *AllAttribute[PAR_NATT_TOTAL] )
+{
+
+   if ( MPI_Rank == 0 )    Aux_Message( stdout, "%s ...\n", __FUNCTION__ );
+
+   const real delta_p[3] = { amr->BoxSize[0]/(Advect_NPar[0]+1), 
+                             amr->BoxSize[1]/(Advect_NPar[1]+1),
+                             amr->BoxSize[2]/(Advect_NPar[2]+1) };
+
+   const real delta_box = amr->BoxSize[0] / MPI_NRank;
+
+   const long NParZ_Local = Advect_NPar[2] / MPI_NRank;
+   const long NPar_ToAssign = Advect_NPar[0]*Advect_NPar[1]*NParZ_Local;
+
+   if ( NPar_ToAssign != NPar_ThisRank )
+      Aux_Error( ERROR_INFO, "total number of particles found in disk [%ld] != expect [%ld] !!\n",
+                 NPar_ToAssign, NPar_ThisRank );
+
+   MPI_Barrier( MPI_COMM_WORLD );
+
+// Generate particles within the uniform sphere of high energy
+
+   for (long kk=0; kk<NParZ_Local;    kk++) 
+   for (long jj=0; jj<Advect_NPar[1]; jj++) 
+   for (long ii=0; ii<Advect_NPar[0]; ii++) 
+   {
+
+      const long p = IDX321( ii, jj, kk, Advect_NPar[0], Advect_NPar[1] );
+
+      ParMass[p] = 0.0;
+
+      ParPosX[p] = (ii+1)*delta_p[0];
+      ParPosY[p] = (jj+1)*delta_p[1];
+      ParPosZ[p] = (kk+1)*delta_p[2] + MPI_Rank*delta_box;
+
+      ParVelX[p] = Advect_Vel[0];
+      ParVelY[p] = Advect_Vel[1];
+      ParVelZ[p] = Advect_Vel[2];
```

Comment:
It's probably better to assign the particle initial velocity **automatically** (using the same interpolation method in `Par_UpdateTracerParticle.cpp`).